src/reactor.cc

```diff
@@ -0,0 +1,406 @@
+    fuel_refill_policy
+        .Init(this, &tritium_reserve, std::string("Input"), &fuel_tracker,
+              buy_quantity, active_dist, dormant_dist, size_dist)
+        .Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    fuel_refill_policy
+        .Init(this, &tritium_reserve, std::string("Input"), &fuel_tracker,
+              std::string("ss"), reserve_inventory, reserve_inventory)
+        .Set(fuel_incommod);
+  } else {
+    throw cyclus::KeyError("Refill mode " + refuel_mode +
+                           " not recognized! Try 'schedule' or 'fill'.");
+    RecordOperationalInfo("Transaction Error",
+                          "Refill mode " + refuel_mode +
+                              " not recognized! Try 'schedule' or 'fill'.");
+  }
+
+  tritium_sell_policy
+      .Init(this, &tritium_storage, std::string("Excess Tritium"))
+      .Set(fuel_incommod)
+      .Start();
+  helium_sell_policy.Init(this, &helium_storage, std::string("Helium-3"))
+      .Set(he3_outcommod)
+      .Start();
+}
+
+std::string Reactor::GetComp(cyclus::Material::Ptr mat) {
+  std::string comp = "{";
+  cyclus::CompMap c = mat->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+  for (std::map<const int, double>::const_iterator it = c.begin();
+       it != c.end(); ++it) {
+    comp = comp + std::string("{") + std::to_string(it->first) +
+           std::string(",") + std::to_string(it->second) + std::string("},");
+  }
+  comp.pop_back();
+  comp = comp + std::string("}");
+  return comp;
+}
+
+void Reactor::Startup() {
+  double reserve_qty = tritium_reserve.quantity();
+  cyclus::Material::Ptr initial_reserve = tritium_reserve.Pop();
+  cyclus::CompMap c = initial_reserve->comp()->atom();
+  cyclus::compmath::Normalize(&c, 1);
+
+  if ((reserve_qty >= (startup_inventory + reserve_inventory)) &&
+      (GetComp(initial_reserve) == "{{10030000,1.000000}}")) {
+    cyclus::Material::Ptr initial_core =
+        initial_reserve->ExtractQty(startup_inventory);
+    tritium_core.Push(initial_core);
+    tritium_reserve.Push(initial_reserve);
+    RecordEvent("Startup", "Core Loaded with " +
+                               std::to_string(tritium_core.quantity()) +
+                               " kg of Tritium.");
+  } else if (GetComp(initial_reserve) != "{{10030000,1.000000}}") {
+    tritium_reserve.Push(initial_reserve);
+    throw cyclus::ValueError(
+        "Startup Failed: Fuel incommod not as expected. " +
+        std::string("Expected Composition: {{10030000,1.000000}}. ") +
+        std::string("Fuel Incommod Composition: ") +
+        std::string(GetComp(initial_reserve)));
+  } else {
+    tritium_reserve.Push(initial_reserve);
+    throw cyclus::ValueError(
+        "Startup Failed: " + std::to_string(tritium_reserve.quantity()) +
+        " kg in reserve is less than required " +
+        std::to_string(startup_inventory + reserve_inventory) +
+        " kg to start-up!");
```

Comment:
Again, my definitions may be off but I thought the amount required for startup was startup_inventory (which includes enough to have a reserve). It's a small difference but may be important so that the model lines up with previous models/variable expectations

Reply:
Agreed, working on a fix to this now. Expect a commit addressing this soon.